torch/distributions.py

```diff
@@ -181,16 +196,34 @@ class Normal(Distribution):
         std (float or Tensor or Variable): standard deviation of the distribution
     """
 
+    has_sample_grad = True
+
     def __init__(self, mean, std):
         self.mean = mean
         self.std = std
 
     def sample(self):
-        return torch.normal(self.mean, self.std)
+    	return torch.normal(self.mean, self.std)
+
+    def rsample(self):
+        eps = torch.normal(torch.zeros_like(self.mean), torch.ones_like(self.std))
```

Comment:
This could be made more efficient by using `new()` and then `normal_()`

Reply:
Will fix this when #4080 is merged

Reply:
Can you please add a TODO then?